src/opnsense/scripts/filter/update_tables.py

```diff
@@ -94,8 +94,20 @@
         # only try to replace the contents of this alias if we're responsible for it (know how to parse)
         if alias.get_parser():
             # when the alias or any of it's dependencies has changed, generate new
-            if alias_changed_or_expired or not os.path.isfile('/var/db/aliastables/%s.txt' % alias_name):
```

Comment:
```
if alias.changed() or not os.path.isfile('/var/db/aliastables/%s.txt' % alias_name):
```

should likely do the trick already.  `changed()` returns `True` if the calculated hash differs and is "sticky" throughout the session. Hashes only exist when there is a parser, which is the same condition as here.  For reference: 

https://github.com/opnsense/core/blob/0e3d660e18c3b7e7fdefc6a9046ced6b60400500/src/opnsense/scripts/filter/lib/alias/__init__.py#L115-L125

https://github.com/opnsense/core/blob/0e3d660e18c3b7e7fdefc6a9046ced6b60400500/src/opnsense/scripts/filter/lib/alias/__init__.py#L195-L205

Writing again on expire when unchanged should indeed not need an additional write, but adding too much glue in already complex code is also something we should prevent. Can you change accordingly and test on your end?